install_process.go

```diff
@@ -99,8 +99,26 @@ func (ip YarnInstallProcess) SetupModules(workingDir, currentModulesLayerPath, n
 		}
 
 	} else {
-		err := os.MkdirAll(filepath.Join(workingDir, "node_modules"), os.ModePerm)
+
+		file, err := os.Lstat(filepath.Join(workingDir, "node_modules"))
+		if err != nil {
+			if !errors.Is(err, os.ErrNotExist) {
+				return "", fmt.Errorf("failed to stat node_modules directory: %w", err)
+			}
+
+		}
+
+		if file != nil && file.Mode()&os.ModeSymlink == os.ModeSymlink {
+			err = os.RemoveAll(filepath.Join(workingDir, "node_modules"))
+			if err != nil {
+				//not tested
```

Comment:
Is a test for this necessary?

Reply:
I don't think so. In fact, the reason I left the comment was that I couldn't seem to find a test case that triggered that failure mode.